src/main/java/jenkins/plugins/slack/decisions/Condition.java

```diff
@@ -0,0 +1,28 @@
+package jenkins.plugins.slack.decisions;
+
+import jenkins.plugins.slack.logging.BuildAwareLogger;
+
+import java.util.function.Predicate;
+import java.util.logging.Logger;
+
+interface Condition extends Predicate<Context> {
+    boolean isMetBy(Context context);
+    boolean userPreferenceMatches();
+    BuildAwareLogger log();
+
+    @Override
+    default boolean test(Context context) {
+        boolean isMet = isMetBy(context);
+        boolean preferences = userPreferenceMatches();
+        if (isMet) {
+            if (preferences) {
+                log().info(context.currentKey(), "will send " + getClass().getSimpleName() + "Notification because build matches and user preferences allow it");
```

Comment:
For some reason the logger is coming up as: `UserAndSystemLogger` and not `Condition`

Any idea why, or if its easily fixable to have the correct class name in the logger?

```
Feb 15, 2019 1:24:55 PM jenkins.plugins.slack.logging.UserAndSystemLogger info
INFO: [freestyle 1 #5] will send OnSuccessNotification because build matches and user preferences allow it
```

Reply:
It's the way jenkins has defined its log formatter: https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/Functions.java#L529

I can update the name to `SlackNotificationsLogger`, which will probably make more sense from the console anyway.